packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,98 +9,319 @@ import 'package:file/file.dart';
+import 'package:flutter_goldens_client/skia_client.dart';
+
 export 'package:flutter_goldens_client/client.dart';
+export 'package:flutter_goldens_client/skia_client.dart';
 
 /// Main method that can be used in a `flutter_test_config.dart` file to set
 /// [goldenFileComparator] to an instance of [FlutterGoldenFileComparator] that
-/// works for the current test.
+/// works for the current test. _Which_ FlutterGoldenFileComparator is
+/// instantiated is based on the current testing environment.
 Future<void> main(FutureOr<void> testMain()) async {
-  goldenFileComparator = await FlutterGoldenFileComparator.fromDefaultComparator();
+  goldenFileComparator = _testingWithSkiaGold(const LocalPlatform())
+    ? await FlutterSkiaGoldFileComparator.fromDefaultComparator()
+    : await FlutterGoldensRepositoryFileComparator.fromDefaultComparator();
   await testMain();
 }
 
-/// A golden file comparator specific to the `flutter/flutter` repository.
+/// Abstract base class golden file comparator specific to the `flutter/flutter`
+/// repository.
+abstract class FlutterGoldenFileComparator implements GoldenFileComparator {
+
+  /// Completes pixel differential tests utilizing [imageBytes] identified by
+  /// [golden].
+  ///
+  /// The returned future completes with a  boolean value that indicates whether
+  /// the imageBytes given matched. The method by which comparison is completed
+  /// is left up to the implementation class. For instance, some
+  /// implementations may load files from the local file system, whereas others
+  /// may upload files over the network.
+  ///
+  /// In the case of comparison mismatch, the comparator may choose to throw a
+  /// [TestFailure] if it wants to control the failure message.
+  @override
+  Future<bool> compare(Uint8List imageBytes, Uri golden);
+
+  /// Updates the golden file identified by [golden] with [imageBytes].
+  ///
+  /// This will be invoked in lieu of [compare] when [autoUpdateGoldenFiles]
+  /// is `true` (which gets set automatically by the test framework when the
+  /// user runs `flutter test --update-goldens`).
+  ///
+  /// The method by which [golden] is located and by which its bytes are written
+  /// is left up to the implementation class.
+  @override
+  Future<void> update(Uri golden, Uint8List imageBytes);
+
+  /// Updates the uri [key] of the golden file to incorporate any [version]
+  /// number.
+  ///
+  /// The [version] is an optional int that can be used to differentiate
+  /// historical golden files.
+  ///
+  /// Version numbers are used in golden file tests for package:flutter. You can
+  /// learn more about these tests [here](https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter).
+  @override
+  Uri getTestUri(Uri key, int version);
+
+  /// Returns the golden file identified by the given [Uri].
+  File getGoldenFile(Uri uri);
+}
+
+/// Implementation of the abstract base class [FlutterGoldenFileComparator] for
```

Comment:
It is kinda an implementation detail that this is "Implementation of the abstract base class [FlutterGoldenFileComparator]". I would omit that, just say: "A [FlutterGoldenFileComparator] for testing golden images against the `flutter/goldens` repository.".